docs/architecture/adr/adr-006-in-cluster-db.md

```diff
@@ -0,0 +1,170 @@
+# ADR 006: In-cluster Database
+
+## Context
+
+We have known, since summer 2024, that Microsoft had decided to sunset Azure
+database for MariaDB. Their recommendation has been to migrate to their Flexible
+MySql database which they have provided pretty well documented. Back then we had
+already experienced database downtime, which prompted us to start a discussion
+about wether to move the databse to Azure Flexible MySql server or to an
+in-cluster database. This conversation has been running for the better part of
+6-8 months.
```

Comment:
I'm not sure all the context of how long we have been discussing this is relevant to the final decision. I would just jump straight into the points.

The main points as to why we are deciding not to follow Azure's recommendation, as I see it, is the stability issues with MariaDB and poor support when things went wrong.

I think the intro would be clearer if it had just these points in order (with your written out wording, of course):

- MariaDB is being sunsetted
- Microsoft recommends MySQL flexible
- Microsoft's operation of MariaDB was very unstable
- Microsoft's support when things went wrong was bad
- So we decided to evaluate alternatives